src/test/resources/data/StudentCourseJoinAuthenticatedTest.json

```diff
@@ -134,6 +134,15 @@
       "team": "Team 1",
       "section": "Section 1"
     },
+    "student1InUnregisteredCourse": {
+      "email": "student1InUnregisteredCourse@gmail.tmt",
+      "course": "idOfUnregisteredCourse",
+      "name": "student1 In unregisteredCourse",
+      "comments": "",
+      "team": "Team 1",
+      "section": "Section 1",
+      "key": "regKeyForStuNotYetJoinCourse"
+    },
```

Comment:
I believe you do not have to add this additional student actually. You can refer to the desired student by using `getTypicalDataBundle()`.